components/camel-zookeeper-master/src/test/java/org/apache/camel/component/zookeepermaster/MasterEndpointIT.java

```diff
@@ -54,14 +55,14 @@ public void testEndpoint() throws Exception {
         List<Route> registeredRoutes = camelContext.getRoutes();
         assertEquals(1, registeredRoutes.size(), "number of routes");
         MasterEndpoint endpoint = (MasterEndpoint) registeredRoutes.get(0).getEndpoint();
-        assertEquals("seda:bar", endpoint.getConsumerEndpointUri(), "wrong endpoint uri");
+        assertEquals("direct:bar", endpoint.getConsumerEndpointUri(), "wrong endpoint uri");
 
         String expectedBody = "<matched/>";
 
         resultEndpoint.expectedBodiesReceived(expectedBody);
 
-        // lets wait for the entry to be registered...
-        Thread.sleep(5000);
+        MasterConsumer masterConsumer = (MasterConsumer) camelContext.getRoute("zookeeper-master-to-direct").getConsumer();
+        Awaitility.await().until(() -> masterConsumer.isMaster() && masterConsumer.isConnected());
```

Comment:
You need some kind of duration to not wait forever, like 10 seconds or so

Reply:
Isn't awaitiltiy default timeout 10 seconds? or am I missing something? https://github.com/awaitility/awaitility/wiki/Usage#simple `By default Awaitility will wait for 10 seconds`

Reply:
Ah you learn something every day. Thanks